main.tf

```diff
@@ -235,7 +236,7 @@ resource "aws_route" "public_internet_gateway_ipv6" {
 ################################################################################
 
 resource "aws_route_table" "private" {
-  count = var.create_vpc && local.max_subnet_length > 0 ? local.nat_gateway_count : 0
+  count = var.create_vpc && local.max_subnet_length > 0 && length(var.private_subnets) > 0 ? local.nat_gateway_count : 0
```

Comment:
The issue here is that, if there are no private subnets (for instance the case you have an internet breakout and you only have public and transit gateway subnets), the private route table is created anyway (but with no subnets attached). However, I have realized this is used in more places in the code and, since I do not want to introduce other issues because of this, I have rolled back this change.

Reply:
Ah you are right, now i see the problem. IMHO outside of the scope of the pull request, maybe we can create separate PR for this fix to no mix things up ? But seriously this is not problem at all :)

Reply:
btw there is already PR https://github.com/terraform-aws-modules/terraform-aws-vpc/pull/543

Reply:
Oops! Good catch. Anyway looks like forgotten (is from last year :-)

Agree, let's not mix things